### ISSUE TITLE
Not working correctly when used for function with different arities

### ISSUE THREAD

**Issue by ludwikbukowski** [arjan/decorator#21]
Example:
```
  @decorate function_result(:ok)
  def testfun(a,b) do a + b end
  @decorate function_result(:ok)
  def testfun(a) do a end
```
Will compile with a warning:
```
warning: this clause cannot match because a previous clause at line XX always matches
```
Example test: 
https://github.com/arjan/decorator/pull/20

**Comment by ludwikbukowski** [comment]
It has been fixed in #20

**Comment by arjan** [comment]
Merged, thanks!
